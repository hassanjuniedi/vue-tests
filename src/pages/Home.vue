<template>
    <div>
        <Status :status="loginStatus"/>
        <hr>
        <login @input="onLoginStatusUpdate" /> 
    </div>
</template>

<script>
import Status from '../components/Status.vue'
import Login from '../components/Login.vue'
export default {
    name: 'Home',
    components: {
        Status,
        Login
    },
    data() {
        return {
        loginStatus: 'idle'
        }
    },
    methods: {
        onLoginStatusUpdate(status) {
        this.loginStatus = status;
    }
  }
}
</script>