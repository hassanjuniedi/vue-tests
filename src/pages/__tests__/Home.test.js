import { mount } from "@vue/test-utils";
import Home from "../Home.vue";

describe('Home page', () => {
    it('render html correctly', () => {
        const wrapper = mount(Home);
        expect(wrapper.html()).toMatchSnapshot();
    })
})