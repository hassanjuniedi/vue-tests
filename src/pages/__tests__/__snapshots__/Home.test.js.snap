// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render html correctly 1`] = `
"<div>
  <div class=\\"idle\\">
    <h1 id=\\"greeting\\">Please login to view you status</h1>
  </div>
  <hr>
  <div class=\\"login\\">
    <div class=\\"form-wrapper\\">
      <form><label for=\\"username\\">
          Username
        </label> <input id=\\"username\\" type=\\"text\\" name=\\"username\\" class=\\"form-control\\"> <br> <label for=\\"password\\">
          Password
        </label> <input id=\\"password\\" type=\\"password\\" name=\\"password\\" class=\\"form-control\\"> <button id=\\"submit\\" type=\\"button\\">Login</button> <button id=\\"reset\\" type=\\"button\\">Reset</button></form>
    </div>
  </div>
</div>"
`;
